x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/slm/SLMSnapshotBlockingIntegTests.java

```diff
@@ -215,6 +215,10 @@ public void testRetentionWhileSnapshotInProgress() throws Exception {
                     SnapshotsStatusResponse s =
                         client().admin().cluster().prepareSnapshotStatus(REPO).setSnapshots(completedSnapshotName).get();
                     assertNull("expected no snapshot but one was returned", s.getSnapshots().get(0));
+                } catch (RepositoryException e) {
+                    // Concurrent status calls and write operations may lead to failures in determining the current repository generation
+                    // TODO: Remove this hack once tracking the current repository generation has been made consistent
```

Comment:
Is there an issue we could link to here so that we have something we can reference to see if it's safe to remove the hack?

Reply:
Not yet I'm afraid. We have https://github.com/elastic/elasticsearch/issues/38941 but the fix to the corruption issue may not fully resolve this situation yet.

Not even sure we have to solve this one with any kind of priority outside of tests:

What's basically happening here is that the snapshot status API breaks for a tiny window during snapshot delete and create (in practice that window will be a little more thanthe latency of one API request). Maybe this is even an ok long term solution? (I'll gather some feedback on that tomorrow and will create an issue accordingly :))

Reply:
We discussed this today during snapshot resiliency sync and I'll code up a fix for this shortly and will remove the todos :)